sdk/inc/azure/core/internal/az_precondition_internal.h

```diff
@@ -117,8 +117,8 @@ AZ_NODISCARD AZ_INLINE bool _az_span_is_valid(az_span span, int32_t min_size, bo
   // left until the theoretical end of the address space, it is not a valid span.
   // Example: (az_span) { .ptr = (uint8_t*)(~0 - 5), .size = 10 } is not a valid span, because most
   // likely you end up pointing to 0x0000 at .ptr[6], &.ptr[7] is 0x...0001, etc.
-  uint8_t* const max_ptr = (uint8_t*)~0;
-  result &= ((size_t)span_size <= (size_t)(max_ptr - ptr));
+  uint8_t* const max_ptr = (uint8_t*)-1;
```

Comment:
it looks weird to use signed numbers for unsigned variables. Can we do left shift to `1` with size of uint8_t?
what's the issue with `~`?   would it work by doing 
```
uint_t const value = ~0;
uint8_t* max_ptr = &value;
```

We don't really need the value to live out of the function right?

Reply:
> what's the issue with `~`?   would it work by doing 

No, it won't.

The issue is with `~0`. We are using a binary negation operation on a signed int `0` (`int` is the  default type for number literals).

I think `~0U` might work. I will try that or use bit-shift instead of negation (however that has the same issue with signing)

Reply:
I recall seeing ~0U used before.  Did it work?

Reply:
> I recall seeing ~0U used before in other code I have worked on outside of Microsoft. Did it work?

Yes, ~0U worked.